Add `yarn rw dev` option to pass any Webpack config as a string

thedavidprice opened the pull request:
Fix #890

This is an alternate, mutual approach to #891 We could merge both and add to CLI docs for `yarn rw dev`

@peterp My approach here is to add a string option to the Yargs command, which could be used to pass any+all Webpack DevServer config options, which will override `webpack.development.js`.

Most importantly --> What do you this of this approach?

Secondarly --> does the code implementation pass your PPSniffTest™?

**edit:** Oh, one question. Should the name of the option be `webpack-config` or `webpackConfig` for CLI option?

peterp commented on packages/cli/src/commands/dev.js:
```diff
@@ -19,6 +19,12 @@ export const builder = (yargs) => {
       description: 'Which dev server(s) to start',
       type: 'array',
     })
+    .positional('webpackConfig', {
+      alias: 'wpc',
+      description:
+        'Optional string of Webpack DevServer config options, e.g. `--wpc="--port=1234"`',
+      type: 'string',
+    })
```
Ooh, I love this... This is what I've wanted for a long time. 

Could we make it more generic? Cause I would like to make this also work for jest, lint, etc.

Maybe something like `--forward`. In yarn and npm this used to just be `--`: `yarn test -- --watchAll`

thedavidprice commented:
## Yargs "Stop Parsing" Forwarder
Yargs does have an option to use `--` to stop parsing and capture remainders into an array: 
https://github.com/yargs/yargs/blob/master/docs/tricks.md#stop-parsing

However, when I try to run using `yarn rw dev -- blahblah`, it appears Yarn intercepts and throws a warning, which seems to disrupt the defaults (i.e. sides: api, web in our case):
```console
warning From Yarn 1.0 onwards, scripts don't require "--" for options to be forwarded. In a future version, any explicit "--" will be forwarded as-is to the scripts.
...
Invalid values:
  Argument: side, Given: "blahBlahblahBlah", Choices: "api", "web"
error Command failed with exit code 1.
```
Bummer ☹️

thedavidprice commented:
@peter this is ready to merge per your approval:
- changed the option name to `forward` with alias `fwd`
- updated description and provided a better example of use

I'm creating a PR with CLI doc update now.

thedavidprice commented:
Doc updated here:
https://github.com/redwoodjs/redwoodjs.com/pull/261

styfle commented:
Thanks! When can we expect a new stable release?

thedavidprice commented:
@styfle By end of today!!
